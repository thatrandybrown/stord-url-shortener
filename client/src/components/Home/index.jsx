import { useState } from "react";
import Form from "../Form";
import NewLink from "../NewLink";

const Home = () => {
  let [data, setData] = useState({});

  return (
    <>
      <Form {...{ setData }} />
      {data && <NewLink {...{ data }} />}
    </>
  );
};

export default Home;
