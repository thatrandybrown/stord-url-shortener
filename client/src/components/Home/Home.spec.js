import { act, fireEvent, render } from "@testing-library/react";
import Home from "./";

jest.mock("../../controllers/shortLink", () => {
  return {
    postShortLink: (url) => {
      if (url === "https://www.google.com")
        return Promise.resolve({
          ok: true,
          json: () => ({
            shortCode: "abcdef",
            target: "https://www.google.com",
          }),
        });
    },
  };
});

test("renders the Form on the Home Page", () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});

test("renders the Form and NewLink text on the Home Page when the form is submitted", async () => {
  const { baseElement, getByLabelText, getByRole } = render(<Home />);
  const urlField = getByLabelText(/URL:/i);
  fireEvent.change(urlField, { target: { value: "https://www.google.com" } });
  const submitButton = getByRole("button", { name: "Shorten" });
  await act(async () => {
    fireEvent.click(submitButton);
  });
  expect(baseElement).toMatchSnapshot();
});
