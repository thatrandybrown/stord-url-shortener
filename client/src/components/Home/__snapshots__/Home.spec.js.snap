// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Form and NewLink text on the Home Page when the form is submitted 1`] = `
<body>
  <div
    id="interstitial-root"
  />
  <div>
    <p>
      Shorten your link: provide a full url and get a shortened link for sharing.
    </p>
    <form>
      <label
        for="url"
      >
        URL:
      </label>
      <input
        id="url"
        name="url"
        required=""
        type="url"
        value=""
      />
      <input
        aria-label="Shorten"
        disabled=""
        name="submit"
        type="submit"
        value="Shorten"
      />
    </form>
    <section
      class="notice"
    >
      <p>
        <a
          href="https://www.google.com"
        >
          https://www.google.com
        </a>
         is now
         
        <a
          class="shortLink"
          href="http://localhost/abcdef"
        >
          http://localhost/
          abcdef
        </a>
      </p>
      <button>
        Copy!
      </button>
    </section>
  </div>
</body>
`;

exports[`renders the Form on the Home Page 1`] = `
<div>
  <p>
    Shorten your link: provide a full url and get a shortened link for sharing.
  </p>
  <form>
    <label
      for="url"
    >
      URL:
    </label>
    <input
      id="url"
      name="url"
      required=""
      type="url"
      value=""
    />
    <input
      aria-label="Shorten"
      disabled=""
      name="submit"
      type="submit"
      value="Shorten"
    />
  </form>
</div>
`;
